dexs/humantap/index.ts

```diff
@@ -0,0 +1,106 @@
+import type { SimpleAdapter } from "../../adapters/types";
+import type { FetchOptions, FetchResultV2, FetchV2 } from "../../adapters/types";
+import { CHAIN } from "../../helpers/chains";
+
+const CONTRACTS = {
+  V1: "0xC9f0Cc4e0cbb1143CcBE12Fc59cE5270112d7845",
+  V2: "0xF92dEC11Eb85DE7a3E618c4E8e4F79Fc60651Ba8",
+  V3: "0x3f117952Bb5FbB19a6aDf4fBDa28F86bC6eD9587",
+  CLAIM: "0xa41C23Dd9525FD8620F3a49305eCecD3639Fbc5c",
+};
+
+const WLD = "0x2cFc85d8E48F8EAB294be644d9E25C3030863003";
+const HTAP = "0xab4EAAC9D4DF861C82A0637db86dE45dd562379a";
+
+const abi = {
+  HTAPBought: "event HTAPBought(address indexed user, uint256 wldSpent, uint256 htapReceived, uint256 timestamp)",
```

Comment:
so, this event is emitted when users purchase on your platform? what percentage of it is used for buyback?